### PR TITLE
feat(grpc): add mayastor v1 service for replica

### PR DESCRIPTION
**Pull request by akhilerm**
- adds the implementation of v1/replica service.
- add tests for v1/replica

### CODE REVIEW

File: mayastor/src/grpc/mayastor/v1.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+            rw_lock: tokio::sync::RwLock::new(None),
+        }
+    }
+}
+
+impl From<Lvs> for Pool {
+    fn from(l: Lvs) -> Self {
+        Self {
+            name: l.name().into(),
+            disks: vec![l.base_bdev().bdev_uri().unwrap_or_else(|| "".into())],
```
Comment (cjones1024) [inline mayastor/src/grpc/mayastor/v1.rs:106]: I'm not sure this is the right thing to do here. What does it mean to have a disk without a name/uri?

Comment (zeenix) [reply mayastor/src/grpc/mayastor/v1.rs:106]: what @cjones1024 said. Maybe TryFrom is more appropriate here? :thinking:

**Comment by jonathan-teh** [comment]
bors try

File: mayastor/src/grpc/v1/replica.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    panic::AssertUnwindSafe,
+    time::Duration,
+};
+use tonic::{Request, Response, Status};
+use ::function_name::named;
+
+#[derive(Debug)]
+pub struct ReplicaService {
+    name: String,
+    interval: Duration,
```
[the remark this answers is not in the document]
Comment (akhilerm) [reply mayastor/src/grpc/v1/replica.rs]: Done.

File: mayastor/src/grpc/v1/replica.rs [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+            share: l.shared().unwrap().into(),
+            uri: l.share_uri().unwrap(),
+        }
+    }
+}
+
+impl ReplicaService {
+    pub fn new(interval: Duration) -> Self {
+        Self {
+            name: String::from("CSISvc"),
```
[the remark this answers is not in the document]
Comment (akhilerm) [reply mayastor/src/grpc/v1/replica.rs]: Done.

**Comment by akhilerm** [comment]
bors try

**Review by dsavitskiy — APPROVED** [review]

**Review by shubham14bajpai — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by akhilerm** [comment]
@gila Can you take a look at this PR?

**Review by mtzaurus — APPROVED** [review]

**Comment by akhilerm** [comment]
bors try

**Comment by jonathan-teh** [comment]
bors try

**Comment by akhilerm** [comment]
bors merge

**Comment by mtzaurus** [comment]
bors merge

**Pull request merged**